lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -246,6 +246,56 @@ class DecomposeAtenAmaxOp : public OpRewritePattern<AtenAmaxOp> {
 };
 } // end namespace
 
+namespace {
+class DecomposeAtenTriuOp : public OpRewritePattern<AtenTriuOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenTriuOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.getSelf();
+    auto inputType = input.getType().dyn_cast<BaseTensorType>();
+    if (!inputType.areAllSizesKnown() || !inputType.hasDtype()) {
+      op->emitError("should have static shape and dtype");
+      return failure();
+    }
+
+    int64_t diagonal;
+    if (!m_TorchConstantInt(&diagonal).match(
+            op.getDiagonal().getDefiningOp())) {
+      op->emitError("non-const diagonal parameter unsupported");
+      return failure();
+    }
+
+    auto inputSize = inputType.getSizes();
+    auto rowNum = *(inputSize.rbegin() + 1);
+    auto colNum = *inputSize.rbegin();
```

Comment:
only support tensor with 2 dimensions?  You should check this like constant diagonal.